### PR TITLE
Tb user story 21

### PR DESCRIPTION
**Pull request by TravisBorgsmiller**
Add user story 21 
 - made user password controller to handle separate logic for user password change
Change test link name for logout in user_logout spec

### CODE REVIEW

File: app/controllers/users_password_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class UsersPasswordController < ApplicationController
+
+  def edit
+
+  end
+
+  def update
+    if params[:password] == params[:password_confirmation]
+        current_user.update(user_params)
+        redirect_to "/profile"
```
Comment (cgaddis36) [inline app/controllers/users_password_controller.rb:10]: Good use of new controller for password updates

File: spec/features/users/edit_password_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    click_button "Log In"
+
+    visit "/profile"
+
+    click_link("Change My Password")
+    expect(current_path).to eq("/user/password/edit")
+
+    password = "new_password"
+
+    fill_in :password, with: password
```
Comment (cgaddis36) [inline spec/features/users/edit_password_spec.rb:50]: Thorough testing

File: spec/features/users/logout_spec.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                                    email: "roman2@example.com",
+                                    password: "hamburger02",
+                                    role: 1
+                                  )
+
+        allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(merchant_user)
+
+        @meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+
+        @tire = @meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
```
Comment (cgaddis36) [inline spec/features/users/logout_spec.rb:49]: Im pretty sure this was merged in on the last pull request... thanks GitHub!

**Review by cgaddis36 — APPROVED** [review]

File: spec/features/users/edit_password_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe "as a registered user visiting profile page" do
+  it "can click a link to change their password" do
+    user = create(:random_user, password: "password", password_confirmation: "password")
```
Comment (adumortier) [inline spec/features/users/edit_password_spec.rb:5]: nice use of factory bot!

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -18,8 +18,8 @@ def create
   end
 
   def show
-    if (session[:user])
-      @user = User.find(session[:user])
+    if current_user
```
Comment (adumortier) [inline app/controllers/users_controller.rb:21]: this allows to pass the tests that fake the user login

File: config/routes.rb [hunk 6]
```diff
@@ -50,4 +50,7 @@
   get "/login", to: "sessions#new"
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy"
+
+  get '/user/password/edit', to: 'users_password#edit'
```
Comment (adumortier) [inline config/routes.rb:54]: the `user_password` controller takes care of the password update and keeps our routes ReSTful

**Review by adumortier — APPROVED** [review]

**Pull request merged**